common/core/desktop/src/km_kbp_debug_api.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * Keyman Core - Public debug API implementation
+ */
+
+#include <cassert>
+#include <algorithm>
+#include <sstream>
+
+#include <keyman/keyboardprocessor.h>
+
+#include "processor.hpp"
+#include "state.hpp"
+
+using namespace km::kbp;
+
+km_kbp_status
+km_kbp_state_debug_set(
+  km_kbp_state *state,
+  int value
+) {
+  assert(state);
+  if(!state) {
+    return KM_KBP_STATUS_INVALID_ARGUMENT;
+  }
+  state->debug_items().set_enabled(value ? true : false);
+  return KM_KBP_STATUS_OK;
+}
+
+uint8_t
+km_kbp_state_debug_get(
+  km_kbp_state const *state
+) {
+  assert(state);
+  if(!state) {
+    return 0;
+  }
+  return state->debug_items().is_enabled() ? 1 : 0;
+}
+
+km_kbp_state_debug_item const *
+km_kbp_state_debug_items(
+  km_kbp_state const *state,
+  size_t *num_items
+) {
+  assert(state && state->debug_items().size() > 0);
+  if (!state || state->debug_items().empty()) {
```

Comment:
Yeah I'm of two minds about this. For shorter content, what I've done here is slightly less clear. But for longer content, e.g. long class names, the style in keyboardprocessor.h is hard to use because the open paren is so far right.

This matches the [style guide](https://github.com/keymanapp/keyman/wiki/Keyman-Code-Style-Guide#code-format) I think and I'm aiming to try and use that pattern. (Disclaimer I just added this format to the style guide 🙈🙉🙊).

Reply:
Haha just in time style guide.   I still think there is nothing wrong with the "If its one line keep it one line rule" 
 ELSE then have the YES  examples on the linked [style guide](url) .   ( keeping the seconde NO)
 This is because of all the functions in xstring.h for example including   `u16ncpy` and all functions similar are just going to blow out and make a lot of noise in the file.

Reply:
See update I just made to style guide